CRM/Utils/Check/Component/DedupeRules.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ * @package CRM
+ * @copyright CiviCRM LLC https://civicrm.org/licensing
+ */
+class CRM_Utils_Check_Component_DedupeRules extends CRM_Utils_Check_Component {
+
+  /**
+   * Get dedupe rules, grouped by contact type, for a specific usage
+   *
+   * @param string $used (Supervised, Unsupervised, General)
+   * @return string[]
+   */
+  private static function getContactTypesForRule($used) {
+    $dedupeRules = \Civi\Api4\DedupeRuleGroup::get()
+      ->addSelect('contact_type')
+      ->setGroupBy([
+        'contact_type',
+      ])
```

Comment:
```suggestion
      ->addGroupBy('contact_type')
```
Sorry for being a nitpicker. My brain is always looking for ways to make code shorter.

Reply:
Looks like the APIv4 Explorer was to blame. This improves it: #22382

Reply:
Ha, I'm the same when I do reviews - my colleagues must find me a nightmare! Pushed this change.